services/web/server/src/simcore_service_webserver/meta_version_control.py

```diff
@@ -0,0 +1,179 @@
+                .set_filter(uuid=str(project_id), prj_owner=self.user_id)
+                .fetch(
+                    [
+                        "type",
+                        "uuid",
+                        "name",
+                        "description",
+                        "thumbnail",
+                        "prj_owner",
+                        "access_rights",
+                        "workbench",
+                        "ui",
+                        "classifiers",
+                        "dev",
+                        "quality",
+                        "published",
+                        "hidden",
+                    ]
+                )
+            )
+            assert project  # nosec
+            project_as_dict = dict(project.items())
+            # FIXME: hack to avoid validation error
+            if "thumbnail" in project_as_dict:
+                project_as_dict["thumbnail"] = project_as_dict["thumbnail"] or ""
+            return project_as_dict
+
+    async def force_branch_and_workcopy(
+        self,
+        repo_id: int,
+        start_commit_id: int,
+        project: ProjectDict,
+        branch_name: str,
+        tag_name: str,
+        tag_message: str,
+    ) -> CommitID:
+        """Forces a new branch with an explicit working copy 'project' on 'start_commit_id'
+
+        For internal operation
+        """
+        IS_INTERNAL_OPERATION = True
+
+        async with self.engine.acquire() as conn:
+            # existance check prevents errors later
+            if tag := await self.TagsOrm(conn).set_filter(name=tag_name).fetch():
+                return tag.commit_id
+
+            # get workcopy for start_commit_id and update with 'project'
+            repo = (
+                await self.ReposOrm(conn).set_filter(id=repo_id).fetch("project_uuid")
+            )
+            assert repo  # nosec
+
+            async with conn.begin():
+
+                # take snapshot of forced project
+                snapshot_checksum = compute_workbench_checksum(project["workbench"])
+
+                # TODO: check snapshot in parent_commit_id != snapshot_checksum
+                await self._upsert_snapshot(
+                    snapshot_checksum, SimpleNamespace(**project), conn
+                )
+
+                # commit new snapshot in history
+                commit_id = await self.CommitsOrm(conn).insert(
+                    repo_id=repo_id,
+                    parent_commit_id=start_commit_id,
+                    message=tag_message,
+                    snapshot_checksum=snapshot_checksum,
+                )
+                assert commit_id  # nosec
+                assert isinstance(commit_id, int)  # nosec
+
+                # creates unique identifier for variant
+                project["uuid"] = eval_workcopy_project_id(
+                    repo.project_uuid, snapshot_checksum
+                )
+
+                # FIXME: File-picker takes project uuid. replace!
```

Comment:
this comment is not very clear? what is the consequence? why do you hide the project as a fix?

Reply:
obsolete

Reply:
All snapshots are hidden so they do not show in the project listing